src/rtop/rtop.ml

```diff
@@ -0,0 +1,36 @@
+let getenv_opt env =
+  try Some(Sys.getenv env)
+  with Not_found -> None
+in
+
+(* Try to read the HOME environment
+ * if it is not set we assume it's windows and fallback to USERPROFILE *)
+let home = match getenv_opt "HOME" with
+| Some(p) -> p
+| None -> Sys.getenv "USERPROFILE"
+in
+
+let () = Unix.openfile (home ^ "/.utoprc") [] 0o640
+|> Unix.close in
+
+let () = Unix.openfile (home ^ "/.utop-history") [] 0o640
+|> Unix.close in
+
+(* Get the dir of the current executable *)
+let dir = Filename.dirname Sys.argv.(0) in
```

Comment:
Copying a suggestion from discord - it's probably better to use `Sys.executable_name` here.  It has some extra logic to try to take advantage of OS-specific methods for determining the actual executable name.  It's not perfect but it is more accurate that `Sys.argv.(0)` in some cases.  https://caml.inria.fr/pub/docs/manual-ocaml/libref/Sys.html#VALexecutable_name